File: frontend/src/lib/models/products/GraphicsCard.ts
import type { ClockSpeed } from "../shared/ClockSpeed";
import type { Memory } from "../shared/Memory";
import type { ProductLong } from "./ProductLong";

export type GraphicsCard = ProductLong & {
  memory: Memory;
  clockSpeed: ClockSpeed;
  busWidth: number;
  cudaCores: number;
  directXSupport: number;
  tdp: number;
};

File: backend/Backend.Domain/Entities/GraphicsCard.cs
using Backend.Domain.ValueObjects;

namespace Backend.Domain.Entities;

public class GraphicsCard : ProductBase
{
    public required Memory Memory { get; init; }
    public required ClockSpeed ClockSpeed { get; init; }
    public required int BusWidth { get; init; }
    public required int CudaCores { get; init; }
    public required int DirectXSupport { get; init; }
    public required int Tdp { get; init; }
}